### PR TITLE
{role} Fix #14782 Update message to not include trailing slash

### LINKED ISSUE #14782: `az role assignment list` breaks with trailing slash
**Describe the bug**

When providing a `--scope` argument to `az role assignment list`, if a trailing slash is accidentally included, the CLI gives a non-descriptive error "Invalid scope. Please use --help to view the valid format."

**To Reproduce**

Using a valid subscription, run e.g.

```
$ az role assignment list --scope /subscriptions/<subscription_id>/resourceGroups/<resource_group>/
Invalid scope. Please use --help to view the valid format.
```

**Expected behavior**

Our customer expected either a clear error or this to remove the trailing slash and work as intended.

When you run the help command, it isn't entirely obvious that a trailing slash is not supported:

```
$ az role assignment list -h
...
    --scope                          : Scope at which the role assignment or definition applies to,
                                       e.g., /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333,
                                       /subscriptions/0b1f6471-1bf0-4dda-
                                       aec3-111122223333/resourceGroups/myGroup, or
                                       /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceG
                                       roups/myGroup/providers/Microsoft.Compute/virtualMachines/myV
                                       M.
```

**Environment summary**

```
$ az --version
azure-cli                         2.10.1

command-modules-nspkg              2.0.3
core                              2.10.1
nspkg                              3.0.4
telemetry                          1.0.4

...

Your CLI is up-to-date.
```

**Additional context**

An Azure Red Hat OpenShift customer encountered this issue when they needed to update role assignments on a VNet and a trailing slash was accidentally included in the commands run. The error provided was not helpful and they were not able to determine the cause of the error without support.

### PR DESCRIPTION
**Pull request by RakeshMohanMSFT**
Fixes https://github.com/Azure/azure-cli/issues/14782

**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
role

File: src/azure-cli/azure/cli/command_modules/role/_params.py [hunk 1]
```diff
@@ -287,7 +287,7 @@ def load_arguments(self, _):
         c.argument('object_type', options_list=['--type', '-t'], help='object type filter, e.g. "application", "servicePrincipal"  "group", etc')
 
     with self.argument_context('role') as c:
-        c.argument('scope', help='scope at which the role assignment or definition applies to, e.g., /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333, /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup, or /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup/providers/Microsoft.Compute/virtualMachines/myVM')
+        c.argument('scope', help='scope at which the role assignment or definition applies to, e.g., /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333, /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup, or /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup/providers/Microsoft.Compute/virtualMachines/myVM without trailing slash.')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/_params.py:290]: **ARM's resource ID is not a URL**, so adding a trailing slash is not a valid usage. There are many other places where an ARM resource ID can be specified, like `--ids` parameter from `az vm show`:

```
> az vm show -h

Command
    az vm show : Get the details of a VM.

Resource Id Arguments
    --ids                  : One or more resource IDs (space-delimited). It should be a complete
                             resource ID containing all information of 'Resource Id' arguments. You
                             should provide either --ids or other 'Resource Id' arguments.
```

Should we include "without trailing slash" in all these help messages? 🤔

**Review by jiasli — CHANGES_REQUESTED** [review]
Need further discussion.

File: src/azure-cli/azure/cli/command_modules/role/_params.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/role/_params.py:290]: A user may even pass `--scope /subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup/providers/Microsoft.Compute/virtualMachines/myVM without trailing slash`, interally!

Comment (RakeshMohanMSFT) [reply src/azure-cli/azure/cli/command_modules/role/_params.py:290]: true, agreed. Let me close the PR then?

Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/role/_params.py:290]: Thanks @RakeshMohanMSFT !

**Pull request closed without merging**